### PR TITLE
bug fix #192 converting spatial type to text during direct connect to…

### LINKED ISSUE #192: UI: Global Data type mapping box not populated in UI when using direct connect
Screenshot 2021-10-07 at 12 36 50 PM
This box is not populated when mysql in direct connect has unsupported data types such as geometry.
However, using the corresponding dump file with unsupported types works as expected.

Sample schema to replicate this:
```

CREATE TABLE `productshapes` (
  `productId` int NOT NULL AUTO_INCREMENT,
  `productLine` varchar(50) NOT NULL ,
  `productShape` geometry,
  `attributes` JSON,
  PRIMARY KEY (`productId`)
) ENGINE=InnoDB DEFAULT CHARSET=latin1;
```

### PR DESCRIPTION
**Pull request by sagarsearce**
### Bug Fix #192 

Two input methods to convert 

#### Method 1

> give the path of the SQL dump file.

#### Method 2

> direct connect with a local database.

as you can see in the [this](https://github.com/cloudspannerecosystem/harbourbridge/issues/192#issuecomment-954445888) issue comment, we were getting different global datatype maps for method 1 and 2, because in input method 1 when the programme encounters some spatial data type during SQL file to source schema conversion it will convert those spatial datatypes into text datatype [here](https://github.com/amar-appdev/harbourbridge/blob/4ca6e5b8da6444aba2f7ee707701e1a012db8ded/mysql/mysqldump.go#L604), but this function is not implemented in input method 2. this pr contains code that will convert any spatial type(geometry, etc) into text type during direct DB connection to source schema conversion.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (anupam-searce) [inline .vscode/launch.json]: Can we add this file to gitignore?

File: mysql/infoschema.go [hunk 2]
```diff
@@ -416,8 +416,22 @@ func getIndexes(conv *internal.Conv, db *sql.DB, table schemaAndName) ([]schema.
...
+		if t == dataType {
+			return true
+		}
+	}
+	return false
+}
+
 func toType(dataType string, columnType string, charLen sql.NullInt64, numericPrecision, numericScale sql.NullInt64) schema.Type {
 	switch {
+	// convert spatial data type into the text
```
Comment (anupam-searce) [inline mysql/infoschema.go]: Can we have this case at last? If the type is not among those basic then only we want to check it for that spatial type.

**Review by anupam-searce — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 1 above)
Comment (sagarsearce) [reply .vscode/launch.json]: added.

File: mysql/infoschema.go (hunk 2 above)
Comment (sagarsearce) [reply mysql/infoschema.go]: done.

**Review by anupam-searce — APPROVED** [review]

**Review by anupam-searce — APPROVED** [review]

**Comment by anupam-searce** [comment]
Looks good

**Comment by sagarsearce** [comment]
there is [another solution](https://github.com/amar-appdev/harbourbridge/pull/4) available that fix the same problem.

**Comment by sagarsearce** [comment]
Direct connect method we can convert spatial type into a string via direct map no need to convert that type into text first. [this](https://github.com/amar-appdev/harbourbridge/pull/4) solution is better than this PR, So closing this one.

**Pull request closed without merging**